src/test/java/net/snowflake/client/jdbc/ProxyLatestIT.java

```diff
@@ -0,0 +1,349 @@
+                            "--proxy-pass-through",
+                            "false", // pass through only matched requests
+                            "--port",
+                            String.valueOf(httpProxyPort),
+                            "--https-port",
+                            String.valueOf(httpsProxyPort),
+                            "--https-keystore",
+                            getResourceURL("wiremock" + File.separator + "ca-cert.jks"),
+                            "--ca-keystore",
+                            getResourceURL("wiremock" + File.separator + "ca-cert.jks"))
+                        .start();
+                waitForWiremock();
+                return true;
+              } catch (Exception e) {
+                return false;
+              }
+            });
+  }
+
+  private void waitForWiremock() {
+    await().atMost(Duration.ofSeconds(5)).until(this::isWiremockResponding);
+  }
+
+  private boolean isWiremockResponding() {
+    try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+      HttpGet request =
+          new HttpGet(String.format("http://%s:%d/__admin/mappings", WIREMOCK_HOST, httpProxyPort));
+      CloseableHttpResponse response = httpClient.execute(request);
+      return response.getStatusLine().getStatusCode() == 200;
+    } catch (Exception e) {
+    }
+    return false;
+  }
+
+  private void stopWiremockStandAlone() {
+    wiremockStandalone.destroyForcibly();
+    await()
+        .alias("stop wiremock")
+        .atMost(Duration.ofSeconds(10))
+        .until(() -> !wiremockStandalone.isAlive());
+    wiremockStandalone.isAlive();
+  }
+
+  private int findFreePort() {
+    try {
+      ServerSocket socket = new ServerSocket(0);
+      int port = socket.getLocalPort();
+      socket.close();
+      return port;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Properties getProperties() {
+    Map<String, String> params = getConnectionParameters();
+    Properties props = new Properties();
+    props.put("host", params.get("host"));
+    props.put("port", params.get("port"));
+    props.put("account", params.get("account"));
+    props.put("user", params.get("user"));
+    props.put("role", params.get("role"));
+    props.put("password", params.get("password"));
+    props.put("warehouse", params.get("warehouse"));
+    props.put("db", params.get("database"));
+    props.put("ssl", params.get("ssl"));
+    props.put("insecureMode", true); // OCSP disabled for wiremock proxy tests
+    return props;
+  }
+
+  private void addProxyProperties(Properties props) {
+    String proxyProtocol = getProxyProtocol(props);
+    props.put("useProxy", "true");
+    props.put("proxyProtocol", proxyProtocol);
+    props.put("proxyHost", WIREMOCK_HOST);
+    props.put("proxyPort", String.valueOf(getProxyPort(proxyProtocol)));
+  }
+
+  private void setJvmProperties(Properties props) {
```

Comment:
There is also a third option - setting it up using env vars. And this is the only option that works with GCP stages if I remember correctly.

Reply:
I think we will need separate task to revisit proxy tests and add missing ones, I started with some base to verify wiremock setup is working

Reply:
but I agree this case should be verified, I will try to add it on Monday if I will have time